Fails to build for me on x86_64-linux.

```
[...]
matrix-dendrite> internal/caching/caches.go:26:27: syntax error: unexpected [, expecting semicolon or newline or }
matrix-dendrite> internal/caching/caches.go:39:14: syntax error: unexpected keyable, expecting ]
matrix-dendrite> internal/caching/caches.go:47:2: syntax error: unexpected ~, expecting method or interface name
matrix-dendrite> internal/caching/impl_ristretto.go:75:48: syntax error: unexpected comma, expecting :
matrix-dendrite> internal/caching/impl_ristretto.go:80:13: syntax error: unexpected : at end of statement
matrix-dendrite> internal/caching/impl_ristretto.go:85:4: syntax error: unexpected comma after top level declaration
matrix-dendrite> internal/caching/impl_ristretto.go:139:38: syntax error: unexpected keyable, expecting ]
matrix-dendrite> internal/caching/impl_ristretto.go:143:39: syntax error: unexpected [, expecting comma or )
matrix-dendrite> internal/caching/impl_ristretto.go:148:32: syntax error: unexpected keyable, expecting ]
matrix-dendrite> internal/caching/impl_ristretto.go:155:33: syntax error: unexpected [, expecting comma or )
matrix-dendrite> internal/caching/impl_ristretto.go:155:33: too many errors
matrix-dendrite> note: module requires Go 1.18
[...]
```

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -22241,10 +22241,7 @@ with pkgs;
 
   dcnnt = python3Packages.callPackage ../servers/dcnnt { };
 
-  dendrite = callPackage ../servers/dendrite {
-    # pinned due to build failure or vendoring problems. When unpinning double check with: nix-build -A $name.go-modules --rebuild
-    buildGoModule = buildGo117Module;
-  };
+  dendrite = callPackage ../servers/dendrite { };
 
   dex-oidc = callPackage ../servers/dex { };
 

